cmd/frontend/internal/usagestats/search.go

```diff
@@ -94,18 +95,52 @@ func searchActivity(ctx context.Context, periodType db.PeriodType, periods int)
 		}
 	}
 
+	searchModeNameToArgumentMatches := map[string]struct {
+		eventName          string
+		argumentName       string
+		getEventStatistics func(p *types.SearchUsagePeriod) *types.SearchEventStatistics
+	}{
+		"plain":       {eventName: "SearchResultsQueried", argumentName: "mode", getEventStatistics: func(p *types.SearchUsagePeriod) *types.SearchEventStatistics { return p.SearchModes.PlainText }},
+		"interactive": {eventName: "SearchResultsQueried", argumentName: "mode", getEventStatistics: func(p *types.SearchUsagePeriod) *types.SearchEventStatistics { return p.SearchModes.Interactive }},
+	}
+
+	for searchMode, eventAndArgNames := range searchModeNameToArgumentMatches {
```

Comment:
Seems cleaner. WDYT?
```suggestion
	for searchMode, match := range searchModeNameToArgumentMatches {
```